pkg/fanal/analyzer/buildinfo/dockerfile.go

```diff
@@ -117,15 +117,38 @@ func parseVersion(nvr string) string {
 	return version
 }
 
-// https://github.com/moby/buildkit/blob/b33357bcd2e3319b0323037c900c13b45a228df1/frontend/dockerfile/dockerfile2llb/convert.go#L474-L482
-func metaArgsToMap(metaArgs []instructions.KeyValuePairOptional) map[string]string {
-	m := make(map[string]string)
+type envGetter struct {
+	m map[string]string
+}
 
-	for _, arg := range metaArgs {
-		m[arg.Key] = arg.ValueString()
+func (e *envGetter) addKeyValuePairsToEnvGetter(kvp instructions.KeyValuePairs) {
+	if e.m == nil {
+		e.m = make(map[string]string)
 	}
 
-	return m
+	for _, kv := range kvp {
+		e.m[kv.Key] = kv.Value
+	}
+}
+
+func metaArgsToEnvGetter(metaArgs []instructions.KeyValuePairOptional) *envGetter {
+	env := &envGetter{
+		m: make(map[string]string),
+	}
+
+	for _, kv := range metaArgs {
+		env.m[kv.Key] = kv.ValueString()
+	}
+	return env
+}
+
+func (e *envGetter) Get(key string) (string, bool) {
```

Comment:
I used logic without `once` (see https://github.com/moby/buildkit/blob/ffadc9ff1531c8573e8522b03611833811f4abdf/frontend/dockerfile/dockerfile2llb/convert.go#L775-L779) to avoid creating new map for each env.